### PR TITLE
[SPARK-35208][SQL][DOCS] Add docs for LATERAL subqueries

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
Add docs for LATERAL subqueries

### Why are the changes needed?
Make SQL Reference complete.

### Does this PR introduce _any_ user-facing change?
Yes

Screen Shot 2022-07-08 at 8 02 36 AM

### How was this patch tested?
Manually build the doc and check.

### CODE REVIEW

**Comment by huaxingao** [comment]
cc @cloud-fan @allisonwang-db

File: docs/sql-ref-syntax-qry-select-lateral-subquery.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+### Description
+
+`LATERAL SUBQUERY` is a subquety that is preceded by the keyword `LATERAL`. It provides a way to cross-reference items in the preceding FROM clause.
```
Comment (HyukjinKwon) [inline docs/sql-ref-syntax-qry-select-lateral-subquery.md:24]:
```suggestion
`LATERAL SUBQUERY` is a subquery that is preceded by the keyword `LATERAL`. It provides a way to cross-reference items in the preceding FROM clause.
```

File: docs/sql-ref-syntax-qry-select-lateral-subquery.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+Without the LATERAL keyword, each subquery is evaluated independently and cannot refer to items in the `FROM` clause. `LATERAL SUBQUERY` makes the complicated
```
Comment (HyukjinKwon) [inline docs/sql-ref-syntax-qry-select-lateral-subquery.md:25]:
```suggestion
Without the `LATERAL` keyword, each subquery is evaluated independently and cannot refer to items in the `FROM` clause. `LATERAL SUBQUERY` makes the complicated
```

File: docs/sql-ref-syntax-qry-select-lateral-subquery.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+### Description
+
+`LATERAL SUBQUERY` is a subquery that is preceded by the keyword `LATERAL`. It provides a way to cross-reference items in the preceding FROM clause.
```
Comment (cloud-fan) [inline docs/sql-ref-syntax-qry-select-lateral-subquery.md:24]:
```suggestion
`LATERAL SUBQUERY` is a subquery that is preceded by the keyword `LATERAL`. It provides a way to cross-reference columns in the preceding FROM clause.
```

File: docs/sql-ref-syntax-qry-select-lateral-subquery.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+Without the `LATERAL` keyword, each subquery is evaluated independently and cannot refer to items in the `FROM` clause. `LATERAL SUBQUERY` makes the complicated
```
Comment (cloud-fan) [inline docs/sql-ref-syntax-qry-select-lateral-subquery.md:25]:
```suggestion
Without the `LATERAL` keyword, subqueroes can only refer to columns in the outer query, but not in the `FROM` clause. `LATERAL SUBQUERY` makes the complicated
```

**Review by cloud-fan — APPROVED** [review]

File: docs/sql-ref-syntax-qry-select-lateral-subquery.md (hunk 4 above)
Comment (huaxingao) [reply docs/sql-ref-syntax-qry-select-lateral-subquery.md]: Fixed. Thanks

File: docs/sql-ref-syntax-qry-select-lateral-subquery.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+### Parameters
+
+* **primary_relation**
+
+  Specifies the primary relation. It can be one of the following:
+  * Table relation
+  * aliased query
+
+    Syntax: `( query ) [ [ AS ] alias ]`
+  * aliased relation
```
Comment (allisonwang-db) [inline docs/sql-ref-syntax-qry-select-lateral-subquery.md:43]:
```suggestion
  * Aliased relation
```

**Review by allisonwang-db — COMMENTED** [review]
Thanks for adding the docs!

File: docs/sql-ref-syntax-qry-select-join.md [hunk 6]
```diff
@@ -26,7 +26,7 @@ A SQL join is used to combine rows from two relations based on join criteria. Th
 ### Syntax
 
 ```sql
-relation { [ join_type ] JOIN relation [ join_criteria ] | NATURAL join_type JOIN relation }
+relation { [ join_type ] JOIN [ LATERAL ] relation [ join_criteria ] | NATURAL join_type JOIN [ LATERAL ] relation }
```
[the remark this answers is not in the document]
Comment (huaxingao) [reply docs/sql-ref-syntax-qry-select-join.md:29]: Sounds good! Changed.

before:

Screen Shot 2022-07-11 at 2 11 21 PM


after:

Screen Shot 2022-07-11 at 2 10 10 PM

**Review by allisonwang-db — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

**Comment by huaxingao** [comment]
Thanks @cloud-fan @allisonwang-db @HyukjinKwon

**Review by dongjoon-hyun — COMMENTED** [review]
+1, LGTM. Thank you all!

**Pull request closed without merging**